### PR TITLE
Stake tests respect MINIMUM_STAKE_DELEGATION

### PR DESCRIPTION
**Pull request by brooksprumo**
#### Problem

If the minimum stake delegation is increased, many of the stake tests break.

#### Summary of Changes

Update the tests to use `MINIMUM_STAKE_DELEGATION`. (Mostly just use/add `MINIMUM_STAKE_DELEGATION` for/to the existing stake amount.)

Also tested locally by changing the minimum stake delegation constant to multiple values: 1, 100_123_456_789.

### CODE REVIEW

File: programs/stake/src/stake_state.rs [hunk 1]
```diff
@@ -2288,7 +2288,7 @@ mod tests {
                 &Authorized::default(),
                 &Lockup::default(),
                 &Rent {
-                    lamports_per_byte_year: 42,
+                    lamports_per_byte_year: stake_lamports,
```
Comment (joncinque) [inline programs/stake/src/stake_state.rs:2291]: nit: what do you think about having these use `Rent::default()` and instead of `42` using the standard rent-exempt amount of `2282880` lamports?  Wherever it makes sense, of course

File: programs/stake/src/stake_state.rs [hunk 2]
```diff
@@ -2351,7 +2351,7 @@ mod tests {
...
```
Comment (joncinque) [inline programs/stake/src/stake_state.rs:2354]: nit: maybe these can be `Rent::default()` instead?

File: programs/stake/src/stake_state.rs [hunk 3]
```diff
@@ -4280,7 +4284,7 @@ mod tests {
         let stake_pubkey = solana_sdk::pubkey::new_rand();
         let rent = Rent::default();
         let rent_exempt_reserve = rent.minimum_balance(std::mem::size_of::<StakeState>());
-        let stake_lamports = rent_exempt_reserve * 3; // Enough to allow half to be split and remain rent-exempt
+        let stake_lamports = rent_exempt_reserve * 3 + MINIMUM_STAKE_DELEGATION * 2; // Enough to allow half to be split and remain rent-exempt
```
Comment (joncinque) [inline programs/stake/src/stake_state.rs:4287]: nit: make this a bit closer to the limits
```suggestion
        let stake_lamports = rent_exempt_reserve * 2 + MINIMUM_STAKE_DELEGATION * 2; // Enough to allow half to be successfully split
```

File: programs/stake/src/stake_state.rs [hunk 4]
```diff
@@ -4380,7 +4384,7 @@ mod tests {
...
```
Comment (joncinque) [inline programs/stake/src/stake_state.rs:4387]: nit: same here, make this a bit closer to the limits
```suggestion
        let stake_lamports = rent_exempt_reserve * 2 + MINIMUM_STAKE_DELEGATION * 2; // Enough to allow half to be successfully split
```

File: programs/stake/src/stake_state.rs [hunk 5]
```diff
@@ -4721,7 +4725,7 @@ mod tests {
         let stake_pubkey = solana_sdk::pubkey::new_rand();
         let rent = Rent::default();
         let rent_exempt_reserve = rent.minimum_balance(std::mem::size_of::<StakeState>());
-        let stake_lamports = rent_exempt_reserve + MINIMUM_STAKE_DELEGATION;
+        let stake_lamports = rent_exempt_reserve + MINIMUM_STAKE_DELEGATION * 2;
```
Comment (joncinque) [inline programs/stake/src/stake_state.rs:4728]: I may be missing something, but this test always splits `stake_lamports`, so I think it should stay as it is

**Review by joncinque — COMMENTED** [review]
Good idea to get this done first!  A few little comments that could make these even better for larger values

File: programs/stake/src/stake_state.rs (hunk 5 above)
Comment (brooksprumo) [reply programs/stake/src/stake_state.rs]: I agree. Done!

File: programs/stake/src/stake_state.rs (hunk 4 above)
Comment (brooksprumo) [reply programs/stake/src/stake_state.rs]: Done!

File: programs/stake/src/stake_state.rs (hunk 3 above)
Comment (brooksprumo) [reply programs/stake/src/stake_state.rs]: Done!

File: programs/stake/src/stake_state.rs (hunk 2 above)
Comment (brooksprumo) [reply programs/stake/src/stake_state.rs]: Done!

File: programs/stake/src/stake_state.rs (hunk 1 above)
Comment (brooksprumo) [reply programs/stake/src/stake_state.rs]: I agree. Done!

**Review by joncinque — DISMISSED** [review]
This looks really great! It's also good to see all those tests that were using `Rent:free` can simply use `MINIMUM_STAKE_DELEGATION`
